### PR TITLE
[10.0][FIX] sale_exception: Add savepoint with assertRaise() + save time in tests

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
This fixes one actual problem (on two) in tests.

@guewen @lmignon @Cedric-Pigeon

**Comment by rousseldenis** [comment]
As record raises is catched by assert, record is in inconsistent state.

**Review by Cedric-Pigeon — APPROVED** [review]
Code Review!

**Comment by rousseldenis** [comment]
@florian-dacosta @hparfr @mourad-ehm as you contributed to this

**Comment by rousseldenis** [comment]
@guewen The last error is due to changes you made in : https://github.com/OCA/server-tools/pull/1647/files

The problem comes calling 'test_all_draft_orders()' method in sale_exception tests.
In base_exception, here : https://github.com/OCA/server-tools/pull/1647/files#diff-c7fd995df65e10a635951020066e82a6R137

reocrds are void and nothing is written (expcetion_ids is void on all sale order records).

Do you have a resolution method or do tests have to be changed ?

We need to solve this as 10.0 branch is red.

**Comment by rousseldenis** [comment]
The 10.0 branch will be fixed as soon as https://github.com/OCA/server-tools/pull/1678 will be merged too.

@pedrobaeza

**Comment by pedrobaeza** [comment]
Merging OCA/server-tools#1678

**Comment by rousseldenis** [comment]
@pedrobaeza :green_apple: !!

**Review by pedrobaeza — APPROVED** [review]

**Review by lmignon — APPROVED** [review]

File: sale_exception/tests/test_sale_exception.py [hunk 1]
```diff
@@ -51,18 +51,16 @@ def test_sale_order_exception(self):
...
-                'order_line': [(0, 0, {'name': p.name,
-                                       'product_id': p.id,
-                                       'product_uom_qty': 2,
-                                       'product_uom': p.uom_id.id,
-                                       'price_unit': p.list_price})]
-            },
-        )
+        with self.assertRaises(ValidationError), self.env.cr.savepoint():
+            so1.write(
+                {
```
Comment (pedrobaeza) [inline sale_exception/tests/test_sale_exception.py]: Is this formatted with black, isn't it? Not very aesthetic and line saving I have to say...

Comment (rousseldenis) [reply sale_exception/tests/test_sale_exception.py]: No it isn't. Maybe in a near future :smile:

Comment (pedrobaeza) [reply sale_exception/tests/test_sale_exception.py]: Ah, then why putting the `{` in a new line instead on the previous, saving one line and, most important, an indentation level?

Comment (rousseldenis) [reply sale_exception/tests/test_sale_exception.py]: I'm in a good mood. Done

Comment (pedrobaeza) [reply sale_exception/tests/test_sale_exception.py]: haha, thanks. It was genuine interest to know if there's a reason for that.

**Comment by rousseldenis** [comment]
@pedrobaeza The coverage(project) lowers because we don't run core Odoo sale tests anymore ...

**Pull request merged**